eventbroadcaster/notifiersink.go

```diff
@@ -29,34 +27,48 @@ type NotifierSink struct {
 	logger   *zap.SugaredLogger
 }
 
-// RegisterNotifierSinks builds notifiers sinks and registers them on the given broadcaster
-func RegisterNotifierSinks(mgr ctrl.Manager, broadcaster record.EventBroadcaster, notifiersConfig eventnotifier.NotifiersConfig, logger *zap.SugaredLogger) (err error) {
-	client := mgr.GetClient()
-
-	notifiers, err := eventnotifier.GetNotifiers(notifiersConfig, logger)
-
+// RegisterNotifierSinks registers notifiers sinks on the given broadcaster
+func RegisterNotifierSinks(mgr ctrl.Manager, broadcaster record.EventBroadcaster, notifiers []eventnotifier.Notifier, logger *zap.SugaredLogger) {
 	for _, notifier := range notifiers {
 		logger.Infof("notifier %s enabled", notifier.GetNotifierName())
 
-		broadcaster.StartRecordingToSink(&NotifierSink{client: client, notifier: notifier, logger: logger})
+		broadcaster.StartRecordingToSink(&NotifierSink{client: mgr.GetClient(), notifier: notifier, logger: logger})
 	}
 
 	corev1Client, _ := corev1client.NewForConfig(mgr.GetConfig())
 
 	broadcaster.StartRecordingToSink(&corev1client.EventSinkImpl{Interface: corev1Client.Events("")})
-
-	return
 }
 
 func (s *NotifierSink) Create(event *corev1.Event) (*corev1.Event, error) {
-	dis, err := s.getDisruption(event)
-	if err != nil {
+	s.logger.Infof("CREATE event received: %s", event.Message)
+
+	var obj client.Object
+
+	switch event.InvolvedObject.Kind {
+	case v1beta1.DisruptionKind:
+		disruption, err := s.getDisruption(event)
+		if err != nil {
+			s.logger.Warn(err)
+			return event, fmt.Errorf("eventnotifier: unable to get Disruption object")
+		}
+
+		obj = &disruption
+	case v1beta1.DisruptionCronKind:
+		disruptionCron, err := s.getDisruptionCron(event)
+		if err != nil {
+			s.logger.Warn(err)
+			return event, fmt.Errorf("eventnotifier: unable to get DisruptionCron object")
```

Comment:
Here I have changed the behavior by returning an error rather `nil`, to retry the event. If we return `nil` the event will not be retried and it will increase the chance to miss events.